### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by SlapDrone**
- merged makefiles
- top level compose for inference/training code (expects .env in repo root)
- split into inference and train modules
- inherit from common cuda/pytorch base image to save space etc
- poetry in train container
- fix python with pyenv -> 3.10.11
- utility scripts for cleaning docker, splitting out "core" dependencies (those used in building train models)

I'm still not super happy with it, gotta get the model caching solution for inference too, and figure out an issue that causes the docker cache to not work for the `pip install transformers[torch]` stage. but i think it's at least better.

### CODE REVIEW

**Comment by SlapDrone** [comment]
You wanna take a look at this at all btw @Lacico or shall I just merge it?

File: Dockerfile.base [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# syntax = docker/dockerfile:1
+FROM nvcr.io/nvidia/pytorch:22.12-py3
```
Comment (Lacico) [inline Dockerfile.base:2]: Is there something restricting us from going to 23.08?

**Review by Lacico — APPROVED** [review]
LGTM - Thanks for getting things organised. I've not tried running it all but it looks solid. There may be more re-runs of the cache but that's ok as that bit of code may get a revamp anyway.

File: Dockerfile.base (hunk 1 above)
Comment (SlapDrone) [reply Dockerfile.base:2]: Nope! Will test if that works too.

**Pull request merged**